modules/saml/lib/Auth/Source/SP.php

```diff
@@ -1137,6 +1137,15 @@ public static function onProcessingCompleted(array $authProcState): void
 
         $state['Attributes'] = $authProcState['Attributes'];
 
+        $config = Configuration::getInstance();
+        $allowUnsolicited = $config->getBoolean('enable.saml20-unsolicited', true);
+
+        Assert::true(
+            $allowUnsolicited,
+            'Unsolicited responsed are denied by configuration.',
+            Error\BadRequest::class,
+        );
```

Comment:
Why is this an Assert and not an exception? And why halt processing completely if the setting is off?